src/Lumi/Components/Responsive.purs

```diff
@@ -1,44 +1,65 @@
 module Lumi.Components.Responsive where
 
 import Prelude
+import Effect (Effect)
+import Effect.Uncurried (EffectFn1, runEffectFn1)
+import Effect.Unsafe (unsafePerformEffect)
+import React.Basic.Hooks (Hook, JSX, ReactComponent, component, element, unsafeHook)
+import React.Basic.Hooks as React
 
-import JSS (JSS, jss)
-import React.Basic (JSX, element)
-import React.Basic.DOM (CSS, css, unsafeCreateDOMComponent)
-
-type ResponsiveProps =
-  { style :: CSS
-  , children :: Array JSX
-  }
-
-mobile :: ResponsiveProps -> JSX
-mobile = element (unsafeCreateDOMComponent "lumi-mobile")
-
-mobile_ :: Array JSX -> JSX
-mobile_ = mobile <<< { style: css {}, children: _ }
-
-desktop :: ResponsiveProps -> JSX
-desktop = element (unsafeCreateDOMComponent "lumi-desktop")
-
-desktop_ :: Array JSX -> JSX
-desktop_ = desktop <<< { style: css {}, children: _ }
-
-styles :: JSS
-styles = jss
-  { "@global":
-      { "lumi-desktop":
-          { "display": "flex"
-          , "flex-shrink": "0"
-          , "@media (max-width: 860px)":
-              { "display": "none !important"
-              }
-          }
-      , "lumi-mobile":
-          { "display": "flex"
-          , "flex-shrink": "0"
-          , "@media (min-width: 861px)":
-              { "display": "none !important"
-              }
-          }
-      }
-  }
+foreign import data MediaQuery :: Type
+
+foreign import data UseMediaQuery :: Type -> Type
+
+useMedia :: String -> Hook UseMediaQuery Boolean
+useMedia mq = unsafeHook (runEffectFn1 useMedia_ mq)
+
+foreign import useMedia_ :: EffectFn1 String Boolean
+
+useIsMobile :: Hook UseMediaQuery Boolean
+useIsMobile = map not (useMedia "(min-width: 860px)")
+
+useIsPhone :: Hook UseMediaQuery Boolean
+useIsPhone = map not (useMedia "(min-width: 448px)")
+
+useIsDesktop :: Hook UseMediaQuery Boolean
+useIsDesktop = useMedia "(min-width: 860px)"
```

Comment:
Can we have these values (`860` and `448`) exported in a module? I think this is somewhat repeated throughout the codebase and if we had them now maybe this is something we could refactor over time.

Reply:
That's probably a good idea.. I just thought the string concat looked messy :sweat_smile: I'd prefer to use these helpers instead of the pixel values where possible though, since we currently are not very consistent on things like min- vs max-. I'll note that in a comment